snuba/web/db_query.py

```diff
@@ -520,17 +525,34 @@ def _get_query_settings_from_config(
     clickhouse_query_settings: MutableMapping[str, Any] = {
         k.split("/", 1)[1]: v
         for k, v in all_confs.items()
-        if k.startswith("query_settings/")
+        if k.startswith(f"{settings_key}/")
     }
 
     if override_prefix:
         for k, v in all_confs.items():
-            if k.startswith(f"{override_prefix}/query_settings/"):
+            if k.startswith(f"{override_prefix}/{settings_key}/"):
                 clickhouse_query_settings[k.split("/", 2)[2]] = v
 
     return clickhouse_query_settings
 
 
+def _get_parallel_read_settings_from_config(
```

Comment:
I don't think you really need this function. The `override_prefix` in snuba is actually the storage name (e.g. `transactions`). If you set `transactions/query_settings/max_parallel_replicas=4` it would only apply that setting to that storage. I believe that's what you want?

Reply:
oh, didn't know that. hm, this means that no code change might be necessary at all. well, except adding a stat for visibility.

Reply:
Ah, no.
Seems that prefix is configured for cluster (and actually seems to be never set) 
and reader inherits it
https://github.com/getsentry/snuba/blob/1b693962ee0bfaea0fdc52c5bc0db80efa0ca0e0/snuba/clusters/cluster.py#L278

https://github.com/getsentry/snuba/blob/1b693962ee0bfaea0fdc52c5bc0db80efa0ca0e0/snuba/clusters/cluster.py#L382

Reply:
@evanh But it seems like a good simplification, can I change what is passed as prefix to dataset?